### PR TITLE
train and evaluate models

### PR DESCRIPTION
**Pull request by MikeLippincott**
This model is trained to predict the AnnexinV feature space in the terminal staining.
The data_splitting strategy is to train on the final timepoint and test on the previous time points.
The model will be trained on the two wells from each dose and tested on the other wells (1).

The model is evaluated on predicting the terminal profiles from each of the previous time points.

Note: The terminal profiles stain for DNA and AnnexinV.

### CODE REVIEW

**Comment by MattsonCam** [comment]
> The data_splitting strategy is to train on the final timepoint and test on the previous time points.

Hey @MikeLippincott, just wanted to address this before finishing the pr review. I would advise against this approach, if possible, to avoid temporal data leakage. [This paper](https://arxiv.org/abs/2207.07048) might help

**Comment by MattsonCam** [comment]
> > The data_splitting strategy is to train on the final timepoint and test on the previous time points.
> 
> Hey @MikeLippincott, just wanted to address this before finishing the pr review. I would advise against this approach, if possible, to avoid temporal data leakage. [This paper](https://arxiv.org/abs/2207.07048) might help

Now that I look at the [data splitting code](https://github.com/WayScience/live_cell_timelapse_apoptosis_analysis/blob/328c54ca7e783b3e6221a220d7f733d66acf6e74/5.bulk_timelapse_model/scripts/0.data_splits.py) from the data splitting pr I've noticed that the train and test data seem to come from using one time point by filtering `bulkdf` (line 35 in the referenced code). Not sure if it makes a difference, but wanted to let you know.

File: 5.bulk_timelapse_model/scripts/1.train_model.py [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+        df_shuffled[col] = np.random.permutation(df_shuffled[col])
+    metadata = df[metadata_columns]
+    df.drop(columns=metadata_columns, inplace=True)
+    X = df.drop(columns=y_columns)
+    y = df[y_columns]
+
+    metadata_shuffled = df_shuffled[metadata_columns]
+    df_shuffled.drop(columns=metadata_columns, inplace=True)
+    X_shuffled = df_shuffled.drop(columns=y_columns)
+    y_shuffled = df_shuffled[y_columns]
```
Comment (MattsonCam) [inline 5.bulk_timelapse_model/scripts/1.train_model.py:66]: Consider separating the shuffling functionality from the separating functionality to stay DRYer

File: 5.bulk_timelapse_model/scripts/1.train_model.py [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+model.fit(train_X, train_y)
+# save the model
+
+model_file_path = pathlib.Path("../models/multi_regression_model.joblib").resolve()
+joblib.dump(model, model_file_path)
+
+
+shuffled_model = MultiOutputRegressor(
+    RandomForestRegressor(
+        n_estimators=1000,
```
Comment (MattsonCam) [inline 5.bulk_timelapse_model/scripts/1.train_model.py:126]: Maybe you've done this somewhere else, but consider mentioning why you choose this model and possibly why you choose this hyperparameter value

File: 5.bulk_timelapse_model/scripts/1.train_model.py [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+        random_state=0,
+    )
+)
+shuffled_model.fit(train_shuffled_X, train_shuffled_y)
```
Comment (MattsonCam) [inline 5.bulk_timelapse_model/scripts/1.train_model.py:130]: Consider using cross-validation to try improving model performance

**Review by MattsonCam — APPROVED** [review]
This LGTM, good job @MikeLippincott!

File: 5.bulk_timelapse_model/scripts/1.train_model.py (hunk 1 above)
Comment (MikeLippincott) [reply 5.bulk_timelapse_model/scripts/1.train_model.py:66]: separated out!

File: 5.bulk_timelapse_model/scripts/1.train_model.py (hunk 3 above)
Comment (MikeLippincott) [reply 5.bulk_timelapse_model/scripts/1.train_model.py:130]: added in cross validation with a 5 fold split - might adjust the split though!

File: 5.bulk_timelapse_model/scripts/1.train_model.py (hunk 2 above)
Comment (MikeLippincott) [reply 5.bulk_timelapse_model/scripts/1.train_model.py:126]: added documentation
